### DISCUSSION TITLE
SSR  CACHING external data feature

### DISCUSSION THREAD

**Discussion by gitsupi** [vercel/next.js#22559] [Ideas]
**Describe the feature you'd like to request**

next js has not cach mechanism for ssr
the external data must be cachable to avoid fech data eech request

**Describe the solution you'd like**

you can add redis key value and a socket to get updates from api points to update redis and ssr page data

**Describe alternatives you've considered**

....

**Comment by gitsupi** [comment]
???

**Comment by stigkj** [comment]
Have you looked into Next.js' static generation (SSG)? As long as you have access to the data from your build environment you can use a `getStaticProps` method to fetch this data and pretender your pages on build. If you don't have access to this data in your build environment, you can use incremental static generation (ISG) which will fetch data, render and cache the page on first access.
